File: src/Server/base/receta_diaComida.js
const express = require('express');
const multer = require('multer');
const fs = require('fs');
const path = require('path');
const db = require('./connection');
const Joi = require("joi");
const router = express.Router();

// Esquema de validación con Joi
const ingredienteSchema = Joi.object({
  Id_Recetas_Dia: Joi.number().required(),
  id_receta: Joi.number().required(),
  Id_Usuario_Alta: Joi.number().required()
});
// Esquema de validación con Joi
const eliminarSchema = Joi.object({
  Id_Recetas_Dia: Joi.number().required(),
  Id_Usuario_Modif: Joi.number().required()
});
// Actualizar receta de desayuno
router.put("/:Id_Recetas_Dia", (req, res) => {
    const { Id_Recetas_Dia } = req.params;
    const { id_receta, Id_Usuario_Alta } = req.body;

  // Validar entrada (fusionando params y body)
  const { error } = ingredienteSchema.validate({
    Id_Recetas_Dia: Number(Id_Recetas_Dia), // Convertir a número
    id_receta,
    Id_Usuario_Alta
  });

  if (error) {
    console.log("Error al validar:", error);
    return res.status(400).json({ error: error.details[0].message });
  }

    const query = `
      UPDATE recetas_dia
      SET Id_Receta_Comida = ?, Id_Usuario_Modif = ?, Fecha_Modif = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [id_receta, Id_Usuario_Alta, Id_Recetas_Dia], (err, result) => {
      if (err) {
        
        console.error("Error al actualizar comida:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Comida actualizado con éxito" });
    });
  });
  
// Eliminar (borrar lógico) receta de desayuno
router.put("/borrar/:Id_Recetas_Dia", (req, res) => {
  const { Id_Recetas_Dia } = req.params;
  const { Id_Usuario_Modif } = req.body;
  
// Validar entrada (fusionando params y body)
const { error } = eliminarSchema.validate({
  Id_Recetas_Dia: Number(Id_Recetas_Dia), // Convertir a número
  Id_Usuario_Modif
});

if (error) {
  console.log("Error al validar:", error);
  return res.status(400).json({ error: error.details[0].message });
}
  
    const query = `
      UPDATE recetas_dia
      SET Id_Receta_Comida = NULL, Id_Usuario_Modif = ?, Fecha_Modif = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [Id_Usuario_Modif, Id_Recetas_Dia], (err, result) => {
      if (err) {
        console.error("Error al eliminar comida:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Comida eliminado" });
    });
  });
  
  
module.exports = router;

File: src/serverP/base/receta_diaComida.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
import mysql.connector
from .connection import get_db_connection

# Crear el router con prefijo y etiquetas
recetas_dia_comida_router = APIRouter()

# Modelos de validación con Pydantic
class IngredienteSchema(BaseModel):
    id_receta: int
    Id_Usuario_Alta: int

class EliminarSchema(BaseModel):
    Id_Usuario_Modif: int

# Actualizar receta de comida
@recetas_dia_comida_router.put("/{Id_Recetas_Dia}")
def actualizar_comida(Id_Recetas_Dia: int, data: IngredienteSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Comida = %s, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.id_receta, data.Id_Usuario_Alta, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Comida actualizada con éxito"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()

# Eliminar (borrado lógico) receta de comida
@recetas_dia_comida_router.put("/borrar/{Id_Recetas_Dia}")
def eliminar_comida(Id_Recetas_Dia: int, data: EliminarSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Comida = NULL, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.Id_Usuario_Modif, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Comida eliminada"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()
